android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModule.java

```diff
@@ -95,25 +95,27 @@ private enum MimeType {
     }
   }
 
-  private PermissionListener webviewFileDownloaderPermissionListener = new PermissionListener() {
-    @Override
-    public boolean onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
-      switch (requestCode) {
-        case FILE_DOWNLOAD_PERMISSION_REQUEST: {
-          // If request is cancelled, the result arrays are empty.
-          if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-            if (downloadRequest != null) {
-              downloadFile();
+  private PermissionListener getWebviewFileDownloaderPermissionListener(String downloadMessage) {
+    return new PermissionListener() {
+      @Override
+      public boolean onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        switch (requestCode) {
+          case FILE_DOWNLOAD_PERMISSION_REQUEST: {
+            // If request is cancelled, the result arrays are empty.
+            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+              if (downloadRequest != null) {
+                downloadFile(downloadMessage);
+              }
+            } else {
+              Toast.makeText(getCurrentActivity().getApplicationContext(), "Cannot download files as permission was denied. Please provide permission to write to storage, in order to download files.", Toast.LENGTH_LONG).show();
```

Comment:
Oh dang.... somehow I missed that this string was also hardcoded. 😞 

Probably need the same solution here... although it could be a separate PR... and boy this is a much more complicated string to have to localize...